src/diffusers/utils/testing_utils.py

```diff
@@ -1377,3 +1380,82 @@ def find_expectation(self, key: DeviceProperties = (None, None)) -> Any:
 
     def __repr__(self):
         return f"{self.data}"
+
+
+# This is for Python 3.8/3.9 support of ignore_cleanup_errors
+
+GenericAlias = List[int]
+
+
+class TemporaryDirectory:
```

Comment:
@sayakpaul See here. This is the 3.10 version of `TemporaryDirectory` with a few changes for 3.8 support. We are using this over `tempfile.TemporaryDirectory`